app/code/Magento/InventorySales/Plugin/InventoryReservationsApi/PreventAppendReservationOnNotManageItemsInStockPlugin.php

```diff
@@ -40,6 +59,11 @@ public function __construct(GetStockItemConfigurationInterface $getStockItemConf
      */
     public function aroundExecute(AppendReservationsInterface $subject, \Closure $proceed, array $reservations)
     {
+        $storeId = $this->storeResolver->getCurrentStoreId();
```

Comment:
The scope of this option is `Website` but not `Store`
So we can resolve website via stock id (reservation contains a stock id)

Also, we can not operate with the current store because this operation (saving reservations) is agnostic to scope